File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
  def removeDuplicates(self, nums: List[int]) -> int:
    l, r, initLen = 0, 0, len(nums)
    while r < initLen:
      while r < initLen and nums[r] == nums[l]:
        r += 1
      if r < initLen:
        nums[l+1] = nums[r]
      l += 1
    return l

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
  length, l, r := len(nums), 0, 0
  for r < length {
    for r < length && nums[r] == nums[l] {
      r++
    }
    if r < length {
      nums[l+1] = nums[r]
    }
    l++
  }
  return l
}